src/model/game/GameManager.ts

```diff
@@ -836,23 +26,54 @@ export default class GameManager {
 
     // Used for loading game specific settings before game is selected.
     static get defaultGame(): Game {
-        return this._gameList.find(value => value.internalFolderName === "RiskOfRain2")!;
+        return this.gameList.find(value => value.internalFolderName === "RiskOfRain2")!;
     }
 
     static get gameList(): Game[] {
-        return [...this._gameList];
+        return EcosystemSchema.r2mm_games.map((game) => {
+            const r2mm = game.r2modman!;
+
+            // Map distribution values onto the r2mm StorePlatformMetadata type.
+            // The `identifier` property may not be present, so we check it beforehand to avoid JSON import lint errors.
+            const distributions = game.distributions.map((x: any) => new StorePlatformMetadata(
+                    GetStorePlatformFromName(x.platform),
+                    x.identifier!
+                )
+            );
+
+            const displayMode = (GameSelectionDisplayMode as any)[(r2mm.gameSelectionDisplayMode as string).toUpperCase()];
+            const instanceType = (GameInstanceType as any)[(r2mm.gameInstanceType as string).toUpperCase()];
+            const installerVariant = GetInstallerVariantFromName(r2mm.packageLoader) as PackageLoader;
+
+            return new Game(
+                game.meta.displayName,
+                r2mm.internalFolderName as GAME_NAME,
+                r2mm.settingsIdentifier as string,
+                r2mm.steamFolderName as string,
+                r2mm.exeNames as string[],
+                r2mm.dataFolderName as string,
+                r2mm.packageIndex as string,
```

Comment:
If things are working correctly these `as` castings shouldn't be needed. In fact seems this hides a bug on `dataFolderName`. Since it can be null in the data but is required for Game object, we should address this problem somehow.

Reply:
The compiler infers these types to be `string | undefined`, so casting was a quick and dirty solution. The JSON schema (and validation in ecosystem-schema) affirm that these types are not undefined, but I have no idea how to tell the compiler that. 

I feel like this has been a saga of me trying and failing to have a conversation with the compiler.

Reply:
Well it seems like the only `string | undefined` property is `r2mm.dataFolderName`, so I'll remove all of the unnecessary ones. Weird

Reply:
The `r2mm_games` getter don't check whether `r2mm.dataFolderName` is `string` or `null` so using `as` here can hide some bugs. I'd get rid of it as well and actually check the value either at the filter in getter or here before it's used (perhaps the latter if the game can be supported without the attribute).